mteb/encoder_interface.py

```diff
@@ -151,3 +156,56 @@ def convert_conv_history_to_query(conversations: Sequence[Sequence[str]]) -> str
             The query.
         """
         ...
+
+
+class ImageEncoder:
+    """Interface for image encoder designed based on VLM2VecWrapper.
+    There is not a perfect 1-1 match, e.g. device can be None here.
+    The intention here is to define the current interface and adapt to as close to MTEB as possible
+    and align as much as possible with sentencetransformers.
+    """
+
+    def __init__(
+        self,
+        device: str | None,
+        **kwargs: Any,
+    ):
+        pass
+
+    def encode(  # current a 1-1 match with Encoder.encode
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        pass
+
+    def get_image_embeddings( # Seems like sentence transformers use a singular encode for both images and text. Not sure if we want to do the same.
+        # If not it might be ideal to redefine Encoder.encode
+        self,
+        images: list[Image.Image] | DataLoader,
+        *, 
+        **kwargs,
+        # removed batch_size, it is not required that it will accept kwargs
+    ) -> np.ndarray:  # added standard output (I believe we actually expect tensors in the code, but would like to be consistent)
+        pass
+
+    def get_text_embeddings( # any reason for this?
+        self,
+        texts: list[str],
+        *, 
+        **kwargs, 
+    ) -> np.ndarray:
+        pass
+
+    def get_fused_embeddings( # hmm what if I have a document with images at specific positions?
+        self,
+        texts: list[str] | None = None,
+        images: list[Image.Image] | DataLoader | None = None, # the requirement for these two to be the same seems odd (docs without images, images without associated text, docs with multiple images)
```

Comment:
A solution here might to do something like:

documents: list[Document]

Where you could then do:

```
images = doc.images
text = doc.text

img_embbeddings = model.encode(images)
text_embeddings = model.encode(text)
# aggregate and return
```

This would also solve the weird cases we have for documents in the corpus with a "title". We could simply convert these to documents as well.

Reply:
yep exactly that - I would say if we want to load it in a low-memory fashion, we would have `Iterable[Document]` - I believe that works with dataloaders?